app/src/main/java/com/github/libretube/ui/fragments/PlayerFragment.kt

```diff
@@ -1040,6 +1036,119 @@ class PlayerFragment : BaseFragment(), OnlinePlayerOptions {
         }
     }
 
+    private fun setupDescription(
+        descTextView: TextView,
+        description: String,
+        exoPlayer: ExoPlayer
+    ) {
+        // detect whether the description is html formatted
+        if (description.contains("<") && description.contains(">")) {
+            descTextView.movementMethod = LinkMovementMethod.getInstance()
+            descTextView.text = HtmlParser.createSpannedText(
+                description,
+                LinkHandler { link ->
+                    // check if the link is a youtube link
+                    if (link.contains("youtube.com") || link.contains("youtu.be")) {
+                        // check if the link is a video link
+                        val videoId = getVideoIdIfVideoLink(link)
+                        if (!videoId.isNullOrEmpty()) {
+                            // check if the video is the current video
+                            if (videoId == this.videoId) {
+                                // get the time from the link
+                                val time = link.substringAfter("t=").substringBefore("&")
+                                // check if the time is valid
+                                if (time.isNotEmpty() && time != link) {
+                                    // get the time in seconds
+                                    val timeInSeconds = getTimeInMillis(time)
+                                    if (timeInSeconds != -1L) {
+                                        // seek to the time
+                                        exoPlayer.seekTo(timeInSeconds)
+                                    }
+                                }
+                                // youtube link without time
+                                // open new player
+                                playNextVideo(videoId)
+                            } else {
+                                // not the current video
+                                // open new player
+                                playNextVideo(videoId)
+                            }
+                        } else {
+                            // not a video link, might be channel or playlist link
+                            // handle normally
+                            val intent = Intent(Intent.ACTION_VIEW, Uri.parse(link))
+                            startActivity(intent)
+                        }
+                    } else {
+                        // not a youtube link
+                        // handle normally
+                        val intent = Intent(Intent.ACTION_VIEW, Uri.parse(link))
+                        startActivity(intent)
+                    }
+                }
+            )
+        } else {
+            // Links can be present as plain text
+            descTextView.autoLinkMask = Linkify.WEB_URLS
+            descTextView.text = description
+        }
+    }
+
+    private fun getVideoIdIfVideoLink(link: String): String? {
```

Comment:
That's not needed, instead we should first parse the URL with `Uri#parse` and then use `uri.getQueryParameter` instead of manually splitting strings